packages/service-provider-core/src/uri-generator.ts

```diff
@@ -41,17 +41,36 @@ const DEFAULT_PORT = '27017';
  */
 const CONFLICT = 'cli-repl.uri-generator.no-host-port';
 
+/**
+ * Invalid host message.
+ */
+const INVALID_HOST = 'cli-repl.uri-generator.invalid-host';
+
 /**
  * Validate conflicts in the options.
  *
  * @param {CliOptions} options - The options.
  */
-function validateConflicts(options: CliOptions): any {
+function validateConflicts(options: CliOptions): void {
   if (options.host || options.port) {
     throw new MongoshInvalidInputError(i18n.__(CONFLICT), CommonErrors.InvalidArgument);
   }
 }
 
+/**
+ * Perform basic validation of the --host option.
+ *
+ * @param {string} host - The value of the --host option.
+ */
+function validateHost(host: string): void {
+  const invalidCharacter = host.match(/[^a-zA-Z0-9\.-:\[\]]/);
```

Comment:
Is the `-` mis-treated in the regex here? I think it should be at the end, right? And we can skip escaping of `.` inside the list.
```suggestion
  const invalidCharacter = host.match(/[^a-zA-Z0-9.:\[\]-]/);
```

Reply:
you are right, yes :) thank you!